### ISSUE TITLE
Use term "natural language" instead of "human language" in glossary?

### ISSUE THREAD

**Issue by himorin** [w3c/wcag#2639]
WCAG 2.2 spec defines a term 'human language' in glossary, which is similar to 'natural language' in i18n glossary

https://w3c.github.io/wcag/guidelines/22/#dfn-human-language-s
> human language
> language that is spoken, written or signed (through visual or tactile means) to communicate with humans 

https://www.w3.org/TR/i18n-glossary/#def_natural_language
> Natural Language (sometimes just language) refers to the spoken, written, or signed communications used by human beings. See also localizable text and syntactic content. 

i18n would like to hear feeling whether to replace with more generally used term "natural language".

**Comment by bruce-usab** [comment]
+1 for natural language but does anyone know how people who advocate for Plain Language distinguish between programming languages and human languages?  My first impression is that "natural language" is comp sci jargon.

**Comment by alastc** [comment]
It is before my time (WCAG 2.0), but I assume it was chosen as a better way of differentiating from programming languages. Or perhaps it raises the question: What is an un-natural language? Human vs programming seems more concrete to me.

It seems like a small change, but it would have a ripple effect to a lot of other documentation, so changing the term is unlikely to get support from the group.

**Comment by bruce-usab** [comment]
Yes, point was to differentiate from programming languages.

Am I correct to that WCAG 2.0 definition predated i18n definition?

**Comment by cstrobbe** [comment]
The term "human language" was chosen instead of "natural language" not primarily to differentiate them from programming languages but for a different reason. The term "natural language" may be defined as "any language that has evolved naturally in humans through use and repetition without conscious planning or premeditation" (I'm quoting [Wikipedia's](https://en.wikipedia.org/wiki/Natural_language) definition for convenience). When I was studying linguistics, a "natural language" was also defined as one that people learn as their native language.
So the term "natural language" excludes constructed languages in which web content may also be created, unless there are people who learn them as their native language. [Esperanto](https://en.wikipedia.org/wiki/Esperanto) is said to have native speakers (though not necessarily as their *only* native language) but most don't (or have very few, such as [Ido](https://en.wikipedia.org/wiki/Ido)). These may seem like edge cases, but their are Wikipedia editions in [Esperanto](https://eo.wikipedia.org/wiki/Vikipedio:%C4%88efpa%C4%9Do) and [Ido](https://io.wikipedia.org/wiki/Frontispico), for example.

WCAG's current definition of "human language" includes those languages. ~~The term "human language" would exclude most constructed languages.~~ **Edit 02.12.2025**: What I meant to write in the last sentence was 'The term "natural language would exclude most constructed languages, unless one adopts a broad definition of "natural language".

**Comment by cstrobbe** [comment]
I have asked the editors of the i18n-glossary to clarify what exactly they mean with (or what they exclude from) their definition of "natural language": [Where does the definition of natural language draw the line between natural languages and other human languages?](https://github.com/w3c/i18n-glossary/issues/16)

**Comment by bruce-usab** [comment]
Thanks @cstrobbe as I think clarification from the the 18n-glossary would be very helpful.

Two examples of constructed human language, in the domain of disability, are [Bliss](https://www.blissymbolics.org/) and [Minspeak](https://minspeak.com/).

Another example of a constructed human language which, this being the Internet, we would want to cover is [Klingon](https://en.wikipedia.org/wiki/Klingon_language).  Which is kind of ironic, since the WCAG term <q>human language</q> would be meant to be applicable a natural language used by non-humans.

**Comment by cstrobbe** [comment]
> Another example of a constructed human language which, this being the Internet, we would want to cover is [Klingon](https://en.wikipedia.org/wiki/Klingon_language). That strikes me as ironic, since the WCAG term "human language" would be meant to be applicable to a natural language used by non-humans.

Believe it or not, but someone on planet Earth actually [spoke only Klingon to his son for three years](https://www.huffpost.com/entry/darmond-speers-dad-spoke_n_363477), so it was a human child's native language for some time.

**Comment by bruce-usab** [comment]
**Task Force response**

Consistent with the reasons provided in this conversation, the WCAG 2 Task Force disagrees with the proposed change.  This decision will be reported to AGWG, but the TF won't be asking approval per se.

**Comment by mbgower** [comment]
I think it's important to not overlook that "language" can describe not just an entire linguistic package for communication  (i.e., Japanese, English) but that it also describes the pieces that make up that language. So we can talk about using "simple language" and that has no bearing on which "mother tongue" we use. 

Thus far the responses to this issue have focused on programming versus 'natural' language, but my larger concern is that many of our SCs require us to differentiate between broad considerations for _human_ languages (i.e., if a video is in German, subtitles in Russian are not fulfilling the core user need of captions) and considerations around grammar, syntax and appropriate language covered by items like Abbreviations, Headings and Labels, etc.

The defined term "human language" gives us the ability to indicate when we are addressing considerations around translation (e.g., Language of Page and Language of Parts).

**Comment by detlevhfischer** [comment]
I have no opinion on this but trust the WCAG 2.X clean-up posse to make the right decision here. Does not seem very important?

**Comment by gundulaniemann** [comment]
Not knowing whether the official definition of the term "natural language" covers this, yet I'd also consider languages used by animals to communicate to each other as 'natural'.

So I see two strong reasons to stick to the term "human language".

Even more, as Klingon has a language key as well (e.g. an ISO 639-2 code of tlh), and I think this is true also for other constructed languages, I rather recommend the i18n colleagues consider changing their term to "human language".

**Comment by GreggVan** [comment]
The problem is that computer languages are also human languages.     So perhaps we need to use **Natural human languages** if you are going to count animal languages as natural languages.

**Comment by patrickhlauke** [comment]
i know this is closed, but referring right back to the original post here...would it perhaps not be a simple fix to add "natural" to the glossary definition? EDIT: maybe even add "constructed" in there if that's a concern...

> human language
> *natural or constructed* language that is spoken, written or signed (through visual or tactile means) to communicate with humans

**Comment by cstrobbe** [comment]
I noticed that I had an error in my comment from [29 September 2022](https://github.com/w3c/wcag/issues/2639#issuecomment-1262497720), which I have now corrected. 

Gundula Niemann: 

> Not knowing whether the official definition of the term "natural language" covers this, ...

As far as I know, there is no official body that provides definitions of linguistic terms for the rest of the world to adopt. The definition of "natural language" in the *The Cambridge Dictionary of Linguistics*, edited by E. K. Brown and J. E. Miller (2013) is:

> 1. A human language like French, English or Zulu spoken by a community, as distinguished from a formal language or a computer language. 

(The second part defines natural language processing.)

The definition implies that there are other human languages, presumably constructed languages, but the dictionary defines neither "human language" nor "constructed language". 

It provides a definition of "artificial language", which does not help us in distinguishing human languages (natural languages plus constructed languages such as Esperanto and Klingon) from computer languages:

> An invented language. Some invented languages, such as Esperanto, were intended to be used as a lingua franca by speakers of different languages. Others were devised for use in logic or computer programming.


Gregg Vanderheiden:

> The problem is that computer languages are also human languages.

This is not how linguists define "human language", as far as I know; they use that term to distinguish that category both from computer languages and animal languages.

Patrick Lauke's suggestion makes perfect sense to me. If there is still concern that some people might think the definition includes computer languages, I'd rather address that in a note.

**Comment by GreggVan** [comment]
@cstrobbe 

I find it hard to understand your last comment. 

you said that there is no definition for human language -- but that it  (presumably the meaning of "human language" ) does not include something.

if there is no definition of the term  -- then one cannot say that X is not included in that term.  

There are examples -- but a list of examples is not definitive -- and if some usages include computer languages as languages  created by humans (and they do) then it is ambiguous.

And given that we do now know that bees and ants have sophisticated languages that are clear and sophisticated. 

Hence the simple expedient of using the term phrase   **_natural human language_** which is unambiguous and avoids all discussion -- and the EASY way to do this is to just add that to the definition  (as @patrickhlauke  suggested)  so we don't have to use a long term everywhere.

**Comment by cstrobbe** [comment]
> you said that there is no definition for human language

Linguists usually don't bother to define "human language"; that term implicitly refers to the languages humans use to communicate with each other on a daily basis. The term they normally use is "natural language" and they mostly use it to distinguish that category of language from both formal languages (such as those used in logic, mathematics and chemistry) and constructed languages. 

I have just now consulted three German lexicons of linguistic terms, in which the definition of "natural language" excludes both constructed languages (which is not what we want) and formal languages (which is desired). (For those who want to know, the three lexicons are the following: *Lexikon der Sprachwissenschaft*, edited by Hadumod Bußmann (2008), *Metzler Lexikon Sprache*, edited by Helmut Glück (2010) and the pocket-sized *Wörterbuch linguistische Grundbegriffe*, edited by Winfried Ulrich (2002). None of these reference works contain an entry for "human language".)

"Natural human language" is not improvement: as soon as you add "natural", you exclude constructed languages, which is undesirable.

--- 

P.S.: The distinction between natural and constructed languages by linguists is not simply for the sake of exclusion. The lexicons mentioned above point out that natural languages, unlike constructed languages, are characterised by changeability (i.e. historical changes), vagueness, and lexical and semantic polysemy.

**Comment by GreggVan** [comment]
@cstrobbe 
what are constructed human languages -- that we want to include?

**Comment by cstrobbe** [comment]
A constructed language or artificial language is "A language deliberately invented by a particular person" (R. L. Trask: *Language and Linguistics: The Key Concepts* (2007)). 

Constructed or artificial languages can be subdivided into three types:

1. **Auxiliary languages** or auxlangs, which are intended for communication between different language communities. The most famous example is [Esperanto](https://en.wikipedia.org/wiki/Esperanto), created by the Polish oculist  L. L. Zamenhof in 1887. Other examples, many of which were inspired by Esperanto, include [Ido](https://en.wikipedia.org/wiki/Ido), [Interlingua](https://en.wikipedia.org/wiki/Interlingua), [Latino sine flexione](https://en.wikipedia.org/wiki/Latino_sine_flexione) and [Volapük](https://en.wikipedia.org/wiki/Volap%C3%BCk). 
2. **Artistic languages** or artlangs, created for use in works of fiction, for example, Klingon (in *Star Trek*), the Elvish languages created by J. R. R. Tolkien and the Dothraki and Valyrian languages developed by David J. Peterson for *Game of Thrones*.
3. **Logical languages** or loglangs, such as Loglang and [Lojban](https://en.wikipedia.org/wiki/Lojban). (Trask writes that these languages "display a fundamental wrong-headedness about what a language is and what it is used for".)

In the context of WCAG, I am primarily interested in auxiliary languages. **Esperanto** is significant enough to have its own Wikipedia version, [two feature films entirely in Esperanto](https://en.wikipedia.org/wiki/Esperanto#Popular_culture) and a number of native speakers (around 1,000 according to Wikipedia, which uses [Ethnologue](https://www.ethnologue.com/language/epo/) as its source; these native speakers would obviously be raised bilingually). 

(Several of the languages mentioned above have ISO language tags: Esperanto has `eo`, Ido has `io`, Volapük has `vo`, Interlingua has `ia`, Lojban has `jbo`, Klingon has `tlh`.)

**Comment by patrickhlauke** [comment]
Proposed PR based on my suggestion above https://github.com/w3c/wcag/pull/4787
